opencensus/utils/monitored_resource_util/monitored_resource.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2018 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+class MonitoredResource(object):
+    """MonitoredResource returns the resource type and resource labels.
+    """
+
+    def get_resource_type(self):
+        """Returns the resource type this MonitoredResource.
+        :return:
+        """
+        raise NotImplementedError
+
+    def get_resource_labels(self):
+        """Returns the resource labels for this MonitoredResource.
+        :return:
+        """
+        raise NotImplementedError
+
+    def _get_resource_label_format(self):
+        return '/g.co/r/{res_type}/'.format(res_type=self.get_resource_type())
+
+
+class GcpGceMonitoredResource(MonitoredResource):
```

Comment:
There should also be a `GcpGaeMonitoredResource` for backwards-compatibility in Python